src/input/parse_json.rs

```diff
@@ -20,8 +22,8 @@ pub enum JsonInput {
     Array(JsonArray),
     Object(JsonObject),
 }
-pub type JsonArray = Vec<JsonInput>;
-pub type JsonObject = LazyIndexMap<String, JsonInput>;
+pub type JsonArray = Arc<SmallVec<[JsonInput; 8]>>;
```

Comment:
I assume we can have more than 8 items right? Is 8 just the default that gets allocated?

Reply:
Yes, how `smallvec` works is you pick a number of entries (in this case 8) which will be stored inline in the struct, and beyond that size it will instead use a heap allocation.